crates/bevy_render/src/lib.rs

```diff
@@ -142,17 +143,19 @@ impl Plugin for RenderPlugin {
             .register_type::<Color>();
 
         if let Some(backends) = options.backends {
-            let windows = app.world.resource_mut::<bevy_window::Windows>();
             let instance = wgpu::Instance::new(backends);
-
-            let surface = windows
-                .get_primary()
-                .and_then(|window| window.raw_handle())
-                .map(|wrapper| unsafe {
-                    let handle = wrapper.get_handle();
-                    instance.create_surface(&handle)
+            let surface = {
+                let windows = app.world.resource_mut::<bevy_window::Windows>();
+                let raw_handle = windows.get_primary().and_then(|window| unsafe {
```

Comment:
Needs a SAFETY comment :)

Reply:
Oh, that's true. The file was missing the SAFETY comment before the change, but I'll add one.

Reply:
Hm, it seems like changing the methods of `Windows` to ensure safety is unpractical, as there are also methods which expose mutable references, which also would need to be unsafe. So I'll make creating `RawHandleWrapper` unsafe instead.

Reply:
> > Making prepare_windows an exclusive system which if I remember correctly would make it run on the main thread
> 
> Ideally we would add a `NonSend` system parameter here, which forces main thread access without blocking parallelism fully.

The marker was already there, I just missed it.